agent/command/papertrail_trace_test.go

```diff
@@ -213,6 +213,184 @@ func TestPapertrailTrace(t *testing.T) {
+		},
+		"wildcard": {
+			files: []testFile{
+				{
+					path:    "test1.tar.gz",
+					content: "foo",
+					want: papertrailTraceFile{
+						filename: "test1.tar.gz",
+						sha256:   "2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae",
+					},
+				},
+				{
+					path:    "test2.tar.gz",
+					content: "bar",
+					want: papertrailTraceFile{
+						filename: "test2.tar.gz",
+						sha256:   "fcde2b2edba56bf408601fb721fe9b5c338d10ee429ea04fae5511b68fbf8fb9",
+					},
+				},
+			},
+			patterns: []string{"*.tar.gz"},
+		},
+		"wildcard-parent": {
+			files: []testFile{
+				{
+					path:    "parent/test1.tar.gz",
+					content: "foo",
+					want: papertrailTraceFile{
+						filename: "test1.tar.gz",
+						sha256:   "2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae",
+					},
+				},
+				{
+					path:    "parent/test2.tar.gz",
+					content: "bar",
+					want: papertrailTraceFile{
+						filename: "test2.tar.gz",
+						sha256:   "fcde2b2edba56bf408601fb721fe9b5c338d10ee429ea04fae5511b68fbf8fb9",
+					},
+				},
+			},
+			patterns: []string{"parent/*.tar.gz"},
+		},
+		"multiple-wildcard-parent": {
+			files: []testFile{
+				{
+					path:    "test-1/parent/test1.tar.gz",
+					content: "foo",
+					want: papertrailTraceFile{
+						filename: "test1.tar.gz",
+						sha256:   "2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae",
+					},
+				},
+				{
+					path:    "test-2/parent/test2.tar.gz",
+					content: "bar",
+					want: papertrailTraceFile{
+						filename: "test2.tar.gz",
+						sha256:   "fcde2b2edba56bf408601fb721fe9b5c338d10ee429ea04fae5511b68fbf8fb9",
+					},
+				},
+			},
+			patterns: []string{"*/parent/*.tar.gz"},
+		},
+	}
+
+	for name, tc := range cases {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			tempdir, err := os.MkdirTemp(os.TempDir(), "papertrail-get-files-test-*")
+			if err != nil {
+				t.Fatalf("make temp dir: %s", err)
+			}
+
+			t.Cleanup(func() { os.RemoveAll(tempdir) })
+
+			for _, f := range tc.files {
+				path := filepath.Join(tempdir, f.path)
+				parent := filepath.Dir(path)
+
+				if err := os.MkdirAll(parent, 0755); err != nil {
+					t.Fatalf("make parent dirs: %s", err)
+				}
+
+				if err := os.WriteFile(path, []byte(f.content), 0644); err != nil {
+					t.Fatalf("write file: %s", err)
+				}
+			}
+
+			got, err := getTraceFiles(tempdir, tc.patterns)
+			if err != nil {
+				t.Fatalf("get trace files: %s", err)
+			}
+
+			if len(got) != len(tc.files) {
+				t.Fatalf("got %d files, expected %d", len(got), len(tc.files))
+			}
+
+			for i, gotf := range got {
+				tcf := tc.files[i]
+
+				if gotf != tcf.want {
+					t.Fatalf("expected '%s', got '%s'", tcf.want, gotf)
+				}
+			}
+		})
+	}
+
+	t.Run("multiple-matches", func(t *testing.T) {
+		tempdir, err := os.MkdirTemp(os.TempDir(), "papertrail-get-files-test-*")
+		if err != nil {
+			t.Fatalf("make temp dir: %s", err)
+		}
+
+		t.Cleanup(func() { os.RemoveAll(tempdir) })
```

Comment:
nit: this can just be replaced with a defer + assert:
```
defer func() {
    assert.NoError(os.RemoveAll(tempdir())
}()
```

Reply:
I doubt Evergreen will ever use Go parallel tests because it would be prohibitively painful to try to make the existing tests work in parallel. If we need to split tests down, we usually rely on Evergreen task organization to split them up.

We've also seen some cases where the removeall fails silently (in cases like Windows because there's an open file), and it's not caught because there's no assertion.

Reply:
Ok! I changed to check the output of RemoveAll for each of these